src/System.Private.CoreLib/shared/System/Runtime/Intrinsics/X86/Avx2.cs

```diff
@@ -546,243 +546,963 @@ public abstract class Avx2 : Avx
         /// __m128i _mm_i32gather_epi32 (int const* base_addr, __m128i vindex, const int scale)
         ///   VPGATHERDD xmm, vm32x, xmm
         /// </summary>
-        public static unsafe Vector128<int> GatherVector128(int* baseAddress, Vector128<int> index, byte scale) => GatherVector128(baseAddress, index, scale);
+        public static unsafe Vector128<int> GatherVector128(int* baseAddress, Vector128<int> index, byte scale)
+        {
+            switch (scale)
+            {
+                case 1:
+                    return GatherVector128(baseAddress, index, 1);
+                case 2:
+                    return GatherVector128(baseAddress, index, 2);
+                case 4:
+                    return GatherVector128(baseAddress, index, 4);
+                case 8:
+                    return GatherVector128(baseAddress, index, 8);
+                default:
+                    throw new ArgumentOutOfRangeException();
```

Comment:
The `scale` parameter has discrete valid values, so the RyuJIT bound-check does not work. Use managed non-const fallback to handle Gather intrinsic, which we do need to worry about compiler optimization on the switch table.

Reply:
Added a comment "The scale parameter should be 1, 2, 4 or 8, otherwise, ArgumentOutOfRangeException will be thrown." for each function header.